CMakeLists.txt

```diff
@@ -404,3 +404,63 @@ message(STATUS "")
 if(NOT (MULTI_CONFIG OR CMAKE_BUILD_TYPE))
   message(WARNING "  CMAKE_BUILD_TYPE is not set; using default settings")
 endif()
+
+option(EMSCRIPTEN "Build with Emscripten" OFF)
```

Comment:
All options should be up at the top -- see the comment. This line should be right below the definition of option(CXX_NEXT). Also, all configure options must be documented. There are comments in the top-level CMakeLists.txt -- search for `*** OPTIONS ***`.